### ISSUE TITLE
Support unmarshalling to `map[string]string`

### ISSUE THREAD

**Issue by ryanc414** [jszwec/csvutil#53]
First of all I'd like to say this is a really great package, really makes up for a deficiency in the standard library!

It would be nice to be able to unmarshal records in a CSV file to a `map[string]string`, for example if we don't know ahead of time which columns to expect. Here is an example usage:

```
var records []map[string]string
if err := csvutil.Unmarshal(input, &records); err != nil {
    return err
}
```

Current output:

```
csvutil: Unmarshal(invalid type *[]map[string]string)
```

How does this sound as a proposal?

**Comment by jszwec** [comment]
Hey, thanks for kind words!

When I was designing this package I made a decision not to support anything else than struct values, because the most common use case is to know at least **some** of the columns. For any extra values that you don't normally expect I added `Unused` method that allows you to handle those. Look at this [example](https://github.com/jszwec/csvutil#unmarshal-and-metadata-) - specifically at field `OtherData`.

To summarize - no, I won't be adding support for maps as destination values, because it seems like an infrequent use case,  adding support for it requires a lot of work and can be already implemented via `Unused()`.
